### ISSUE TITLE
Azure Permissions Missing `Microsoft.Network/publicIPPrefixes/join/action` 

### ISSUE THREAD

**Issue by bennerv** [kubernetes-sigs/cloud-provider-azure#7500]
If you attempt to use the annotation on a service `type: LoadBalancer` in Azure with the annotation `service.beta.kubernetes.io/azure-pip-prefix-id` pointing to an Public IP address prefix, it will fail if you give the identity used for CCM the permissions outlined on the Azure Permissions page.  

dummy service
```
apiVersion: v1
kind: Service
metadata:
  name: my-service
  annotations:
    "service.beta.kubernetes.io/azure-pip-prefix-id": "/subscriptions/<subscription-id>/resourceGroups/<resource-group>/providers/Microsoft.Network/publicIPPrefixes/<resource-name>"
spec:
  selector:
    app.kubernetes.io/name: MyApp
  ports:
    - protocol: TCP
      port: 80
      targetPort: 9376
  type: LoadBalancer
```

error message:
```
7s          Warning   SyncLoadBalancerFailed    service/my-service                                    Error syncing load balancer: failed to ensure load balancer: Retriable: false, RetryAfter: 0s, HTTPStatusCode: 403, RawError: {"error":{"code":"LinkedAuthorizationFailed","message":"The client '<principal-id>' with object id '<principal-id>' has permission to perform action 'Microsoft.Network/publicIPAddresses/write' on scope '/subscriptions/<subscription-id>/resourceGroups/<resource-group>/providers/Microsoft.Network/publicIPAddresses/<public-ip-created-off-of-prefix>'; however, it does not have permission to perform action(s) 'Microsoft.Network/publicIPPrefixes/join/action' on the linked scope(s) '/subscriptions/<subscription-id>/resourceGroups/<resource-group>/providers/Microsoft.Network/publicIPPrefixes/<resource-name>' (respectively) or the linked scope(s) are invalid."}}
```

Principal ID permissions:
```
$ az role assignment list --all --assignee <object-id> 
  {
    "id": "/subscriptions/<subscription-id>/resourceGroups/cadenaro/providers/Microsoft.Authorization/roleAssignments/<role-assignment-id>",
    "principalType": "ServicePrincipal",
    "roleDefinitionId": "/subscriptions/<subscription-id>/providers/Microsoft.Authorization/roleDefinitions/<custom-role-definition>",
    "scope": "/subscriptions/<subscription-id>/resourceGroups/<resource-group>",
    "type": "Microsoft.Authorization/roleAssignments",
  },
...
```

Role Definition:
```
{
  "id": "/subscriptions/<subscription-id>/providers/Microsoft.Authorization/roleDefinitions/<role-definition-id>",
  "properties": {
    "assignableScopes": [
      "/"
    ],
    "permissions": [
      {
        "actions": [
          "Microsoft.Compute/virtualMachines/read",
          "Microsoft.Network/loadBalancers/read",
          "Microsoft.Network/loadBalancers/write",
          "Microsoft.Network/networkInterfaces/read",
          "Microsoft.Network/networkSecurityGroups/read",
          "Microsoft.Network/networkSecurityGroups/write",
          "Microsoft.Network/publicIPAddresses/join/action",
          "Microsoft.Network/publicIPAddresses/read",
          "Microsoft.Network/publicIPAddresses/write",
          "Microsoft.Network/virtualNetworks/subnets/join/action",
          "Microsoft.Network/virtualNetworks/subnets/read",
          "Microsoft.Network/loadBalancers/backendAddressPools/join/action",
          "Microsoft.Network/networkInterfaces/write"
        ],
        "dataActions": [],
        "notActions": [],
        "notDataActions": []
      }
    ],
  },
  "type": "Microsoft.Authorization/roleDefinitions"
}

**Comment by bennerv** [comment]
Page in question: https://cloud-provider-azure.sigs.k8s.io/topics/azure-permissions/

**Comment by bennerv** [comment]
closed because #7501 merged
